Training2/src/task3/src/App.jsx

```diff
@@ -0,0 +1,71 @@
+import { useEffect, useRef, useState } from 'react'
+import './App.css'
+import CountDown from './Component/CountDown/CountDown'
+// import CountdownSet from './Component/CountdownSet/CountdownSet'
+
+function App() {
+  const [countdown, setCountdown] = useState('')
```

Comment:
- Missing callback function for a lot of functions
    - Function clearInterval call many times, but we only should call it ONE time when isRunningCountdown = false
    - useEffect is missing isRunningCountDown so how can you check if(isRunningCountdown && countdown < 0){ work correctly, that means the isRunningCountDown is not up to date when you try to use it
    - You only pass onChange to the input, that does not make sense, you have the value, if you follow controlled & uncontrolled concepts, in control, you need to pass both value & onChange, if you follow the uncontrolled concept, you should not use value & onChange, you are doing the partial thing

Reply:
- 
- Tại có 2 hàm là stop và pause nên em nghĩ là sẽ có 2 hàm clearInterval để chặn lại nên em mới gọi đến nhiều lần.
- Em không hiểu ý chỗ này lắm.
- Em đã sửa lại rồi.

Reply:
2. Rõ ràng tụi nó giống nhau đều là clearInterval khi dừng countdown cụ thể là isRunningCountdown = false, a không thấy lý do gì mình lại để clearInterval ở 2 chỗ
3. E để thiếu isRunningCountDown trong dependencies của useEffect vì vậy câu điều kiện (isRunningCountdown && countdown < 0) sẽ không chạy đúng nữa vì nó không đảm bảo isRunningCountdown mang giá trị mới nhất nữa

Reply:
Dạ em sửa lại 2 cái đó rồi ạ.